client_test.go

```diff
@@ -194,6 +194,33 @@ func TestClient_Do_fails(t *testing.T) {
 	}
 }
 
+func TestClient_Do_doesNotRetryOnURISchemeErrors(t *testing.T) {
```

Comment:
Pretty nit-picky but this could have also been a table test as well. Not a huge deal, I'll let you decide if you want to fold it up or keep the additional test.